web/app/models.py

```diff
@@ -808,3 +808,60 @@ def is_octoprint_connected(self):
         return channels.num_ws_connections(
             channels.octo_group_name(self.printer.id)
         ) > 0
+
+
+class NotificationChannel(models.Model):
+    EMAIL = "email"
+    SMS = "sms"
+    DISCORD = "discord"
+    PUSHOVER = "pushover"
+    PUSHBULLET = "pushbullet"
+    TELEGRAM = "telegram"
+    SLACK = "slack"
+    MOBILE_PUSH = "mobile_push"
+
+    CHANNEL_TYPES = (
+        (EMAIL, EMAIL),
+        (SMS, SMS),
+        (PUSHOVER, PUSHOVER),
+        (PUSHBULLET, PUSHBULLET),
+        (DISCORD, DISCORD),
+        (TELEGRAM, TELEGRAM),
+        (SLACK, SLACK),
+        (MOBILE_PUSH, MOBILE_PUSH),
+    )
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    type = models.CharField(max_length=32, choices=CHANNEL_TYPES)
+    config = models.TextField(default='', blank=True)
+
+    enabled = models.BooleanField(default=True)
+
+    notify_on_failure_alert = models.BooleanField(blank=True, default=True)
+    notify_on_account_events = models.BooleanField(blank=True, default=False)
+
+    notify_on_print_done = models.BooleanField(blank=True, default=False)
+    notify_on_print_cancelled = models.BooleanField(blank=True, default=False)
+    notify_on_filament_change = models.BooleanField(blank=True, default=False)
+    notify_on_other_events = models.BooleanField(blank=True, default=False)
+
+    notify_on_heater_status = models.BooleanField(blank=True, default=False)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        unique_together = ('user', 'type')
+
+    def phone_country_code(self):
```

Comment:
This looks quite hacky. 3 options:

1. make SMS a special case and have it own data model.
2. move this code to the sms channel impl itself.
3. make 2 config fields: `config1` and `config2`. This may be a good option as there could be other channels that will need 2 config fields (such as id + secret)